File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer> hm = new HashMap();
        int ans = 0;
        int start =0;
        int i =0;
        while(i<s.length())
        {
            if (hm.containsKey(s.charAt(i)) && (start<=hm.get(s.charAt(i))))
            {
                start = hm.get(s.charAt(i));
            }
            hm.put(s.charAt(i),i);
            ans = Math.max(ans, i-start);
            i++;
        
        }
        return ans;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-2.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x:x[0])
        ans=[]
        i =0
        while i<len(intervals):
            start = intervals[i][0]
            end = intervals[i][1]
            while i<len(intervals)-1 and intervals[i+1][0]<=end:
                end=max(end,intervals[i+1][1])
                i=i+1
            i=i+1
            ans.append([start,end])
        return ans


